Gini-iOS-SDK/GINIDocumentTaskManager.h

```diff
@@ -132,46 +101,55 @@
+ * Creates a new partial document with the given `doctype` from the given data.
  * Data can be in the format of a PDF, UTF-8 text or image representation.
  * By providing the doctype, Gini's document processing is optimized in many ways.
  *
- * See the [Gini API documentation](http://developer.gini.net/gini-api/html/documents.html#document-type-hints) for
- * details and [a list of available doctypes](http://developer.gini.net/gini-api/html/entity_reference.html#extraction-entity-doctype).
+ * See the [Gini API documentation](Add documentation link).
  *
  * @param fileName                  The file name of the document.
  * @param data                      Data representing the document.
- * @param docType                   The doctype hint for the document.
+ * @param docType                   The doctype hint for the document [Possible values](http://developer.gini.net/gini-api/html/entity_reference.html#extraction-entity-doctype).
  * @param cancellationToken         Cancellation token used to cancel the current task.
  *
  * @returns                         A `BFTask*` that will resolve to a `GINIDocument` instance representing the created document.
  *                                  Please notice that it is very unlikely that the created document is already fully processed, so
  *                                  the extractions may not yet exist.
  */
-- (BFTask *)createDocumentWithFilename:(NSString *)fileName
-                              fromData:(NSData *)data
-                               docType:(NSString *)docType
-                     cancellationToken:(BFCancellationToken *)cancellationToken;
+- (BFTask *)createPartialDocumentWithFilename:(NSString *)fileName
+                                     fromData:(NSData *)data
+                                      docType:(NSString *)docType
+                            cancellationToken:(BFCancellationToken *)cancellationToken;
 
 /**
- * Saves updates on the extractions.
+ * Creates a new composite document
  *
- * Updating extractions is called "Submitting feedback" in the Gini API documentation.
+ * See the [Gini API documentation](Add documentation link).
  *
- * @param document      The document.
+ * @param partialDocumentsInfo  Array containing the partial documents info (document url and additional parameters).
+ * @param fileName              The filename of the document.
+ * @param docType               (Optional) A doctype hint. This optimizes the processing at the Gini API.
+ *                              for a list of possibles doctypes.
+ * @param cancellationToken     Cancellation token used to cancel the current task.
+ *
+ * @returns                     A`BFTask*` that will resolve to a NSString containing the created document's ID.
  */
-- (BFTask *)updateDocument:(GINIDocument *)document;
+
+- (BFTask *)createCompositeDocumentWithPartialDocumentsInfo:(NSArray<NSDictionary<NSString *, id>*>*)partialDocumentsInfo
```

Comment:
I would prefer a more stricter public api. This method is extensible which is great. The downside is, that it's harder to use: what would I as a client have to put into the dictionary?

By narrowing it down to what we need right now would make it imho easier to use. We need a list of documents and rotation deltas.

For example:
```
- (BFTask *)createCompositeDocumentWithPartialDocumentsInfo:(NSArray<PartialDocumentInfo*>*)partialDocumentsInfo
```

`PartialDocumentInfo` could contain the `GINIDocument` (or just the document url) and the rotation delta. Using a container object is more verbose and looks nicer to me than using a dictionary inside of an array. This also maintains the document order like in your implementation.